src/visualizers-workbench/P5VisualizerTemplate.ts

```diff
@@ -0,0 +1,174 @@
+        // storing its value as a local constant for future reference
+        const sketch = this.sketch
+
+        // scale and center the coordinate system. calculations involving the
+        // canvas dimensions need to be done every frame, because the dimensions
+        // can change at any time
+        sketch.translate(0.5 * sketch.width, 0.5 * sketch.height)
+        const smallDim = Math.min(sketch.width, sketch.height)
+        sketch.textSize(0.2 * smallDim)
+
+        // paint the background. for an animated visualizer, this is often done
+        // as the first drawing step in each frame, wiping out the previous
+        // frame and leaving a blank canvas to draw on
+        sketch.background(206, 212, 218)
+
+        // fade the white flash that shows the index has changed, and set the
+        // text fill color accordingly
+        if (this.justStepped) {
+            this.justStepped = false
+        } else {
+            this.flash *= Math.exp(-0.01 * sketch.deltaTime)
+        }
+        if (this.flash < 0.001) {
+            // stop the animation when the flash has faded
+            this.flash = 0
+            sketch.noLoop()
+        }
+        const textFill = sketch.lerpColor(
+            sketch.color(128, 159, 255),
+            sketch.color(255),
+            this.flash
+        )
+        sketch.fill(textFill)
+
+        // print the current element
+        const element = this.seq.getElement(this.index)
+        sketch.text(element.toString(), 0, 0)
+        console.log('drawn')
+    }
+
+    // when the user presses a key while the sketch is in focus, this function
+    // is called to handle the input
+    //
+    //   https://p5js.org/reference/#/p5/keyPressed
+    //
+    // there are similar functions for other input events
+    //
+    //   https://p5js.org/reference/#group-Events
+    //
+    keyPressed() {
+        // accessing `this.sketch` triggers some consistency checks, so it's
+        // most parsimonious to access it at most once per function call,
+        // storing its value as a local constant for future reference
+        const sketch = this.sketch
+
+        const oldIndex = this.index
+        if (
+            sketch.keyCode === sketch.RIGHT_ARROW
+            && this.index < this.seq.last
+        ) {
+            this.index += this.stepSize
+        } else if (
+            sketch.keyCode === sketch.LEFT_ARROW
+            && this.index > this.seq.first
+        ) {
+            this.index -= this.stepSize
+        }
+
+        // show a flash when the index changes. the `loop()` function re-starts
+        // the sketch's animation loop
+        if (this.index !== oldIndex) {
+            this.justStepped = true
+            this.flash = 1
+            sketch.loop()
+        }
+    }
+}
+
+export const exportModule = new VisualizerExportModule(
+    P5VisualizerTemplate, // visualizer class
+    'A template for p5 visualizers' // short description
```

Comment:
This string should describe the actual visualizer being defined here, e.g. `Display sequence entries one at a time`

Reply:
Still unresolved.

Reply:
Nice catch. Done!